### PR TITLE
adding the block production test

### PR DESCRIPTION
**Pull request by ghost-not-in-the-shell**
Thank you for contributing to Mina! Please see `CONTRIBUTING.md` if you haven't
yet. In that doc, there are more details around how to start our CI.

Explain your changes here.

Explain how you tested your changes here.

Checklist:

- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

Closes #0000
Closes #0000

### CODE REVIEW

File: buildkite/scripts/run-test.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+#     docker pull "$coda_daemon_image" && break
+#     [ "$i" != 30 ] || (echo "coda-daemon image never appeared in docker registry" && exit 1)
+#     sleep 60
+# done
+
+echo "--- Building test-executive"
+
+dune build --verbose --profile=testnet_postake_medium_curves src/app/test_executive/test_executive.exe
+
+echo "--- Clone coda-automation repo (this would be changed in future. Nathan is suggesting linking coda-automation in coda repo)"
```
Comment (0x0I) [inline buildkite/scripts/run-test.sh:34]: Just a note: we can remove all of the redundancy below by running the `dune build...` step above once, uploading it as an artifact and then allowing all tests cases to pull it and execute on agents (without each test having to build its own).

File: buildkite/src/Constants/ContainerImages.dhall [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 {
   toolchainBase = "codaprotocol/ci-toolchain-base:v3",
-  codaToolchain = "codaprotocol/mina-toolchain@sha256:d4ad6dec3a846e425f97de55a697de9d65327e76782337fcdcfb373f95d21dd4",
+  codaToolchain = "codaprotocol/mina-toolchain-030ed2ce08cc3d7ee57f7e1942c8278a7cef237f",
```
Comment (0x0I) [inline buildkite/src/Constants/ContainerImages.dhall:5]: Not a huge fan of SHA tagging due mostly to readability concerns tho definitely agree it's the most reliable way to ensure an image isn't changed from underneath you. Anyway, can see keeping the sha but creating and using a `latest` tag could work here.

File: automation/terraform/buildkite/buildkite-ci/us-east4.tf [hunk 3]
```diff
@@ -68,7 +68,7 @@ locals {
         limits = {
           cpu    = "15.5"
           memory = "16G"
-          ephemeral-storage = "100Gi"
+          ephemeral-storage = "500Gi"
```
Comment (0x0I) [inline automation/terraform/buildkite/buildkite-ci/us-east4.tf:71]: Do we still need this bump with the build steps separation?

Comment (ghost-not-in-the-shell) [reply automation/terraform/buildkite/buildkite-ci/us-east4.tf:71]: No, I think the problem is not diskspace limit. I would undo this commit.

**Pull request closed without merging**